modin/experimental/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -0,0 +1,1283 @@
+        Function to be applied to the frame.
+    """
+
+    def caller(query_compiler, *args, **kwargs):
+        df = query_compiler._modin_frame
+        if df_copy:
+            df = df.copy()
+        if is_series:
+            df = df.squeeze(axis=1)
+        exclude_names = [
+            # "broadcast",
+            "fold_axis",
+            # "squeeze_self",
+            # "squeeze_value",
+            "ignore_indices",
+        ] + filter_kwargs
+        kwargs = kwargs.copy()
+        for name in exclude_names:
+            kwargs.pop(name, None)
+        args = try_cast_to_pandas(args, squeeze=squeeze_args)
+        kwargs = try_cast_to_pandas(kwargs, squeeze=squeeze_kwargs)
+        result = func(df, *args, **kwargs)
+        inplace_method = kwargs.get("inplace", False)
+
+        if in_place:
+            inplace_method = in_place
+        if inplace_method:
+            result = df
+        if not (return_modin or isinstance(result, (pandas.Series, pandas.DataFrame))):
+            return result
+        if isinstance(result, pandas.Series):
+            if result.name is None:
+                result.name = MODIN_UNNAMED_SERIES_LABEL
+            result = result.to_frame()
+
+        return query_compiler.__constructor__(result)
+
+    return caller
+
+
+@_inherit_docstrings(BaseQueryCompiler)
+class NativeQueryCompiler(BaseQueryCompiler):
+    """
+    Query compiler for the pandas storage format.
+
+    This class translates common query compiler API into
+    plain pandas to execute operations on small data
+    depending on the threshold.
+
+    Parameters
+    ----------
+    pandas_frame : pandas.DataFrame
+        Pandas frame to query with the compiled queries.
+    """
+
+    def __init__(self, pandas_frame):
+        assert NativeDataframeMode.get() == "Native_Pandas"
+        if hasattr(pandas_frame, "_to_pandas"):
+            pandas_frame = pandas_frame._to_pandas()
+        if is_scalar(pandas_frame):
+            pandas_frame = pandas.DataFrame([pandas_frame])
+        elif not isinstance(pandas_frame, pandas.DataFrame):
+            pandas_frame = pandas.DataFrame(pandas_frame)
+
+        self._modin_frame = pandas_frame
+
+    def execute(self):
+        pass
+
+    @property
+    def frame_has_materialized_dtypes(self) -> bool:
+        """
+        Check if the undelying dataframe has materialized dtypes.
+
+        Returns
+        -------
+        bool
+        """
+        return True
+
+    def set_frame_dtypes_cache(self, dtypes):
+        """
+        Set dtypes cache for the underlying dataframe frame.
+
+        Parameters
+        ----------
+        dtypes : pandas.Series, ModinDtypes, callable or None
+        """
+        pass
+
+    def set_frame_index_cache(self, index):
+        """
+        Set index cache for underlying dataframe.
+
+        Parameters
+        ----------
+        index : sequence, callable or None
+        """
+        pass
```

Comment:
Should we directly assign new index cache?

Reply:
Added comment this is for consistency with other QCs, dtypes should be assigned directly on the frame